### ISSUE TITLE
Jump to symbol doesn't change file where it should

### ISSUE THREAD

**Issue by karolyi** [DamnWidget/anaconda#682]
## Expected Behaviour
Jump to symbol should jump to the symbol definition

## Actual Behaviour
At some occasions, it jumps to the proper line, but in the file where I'm at instead of the file where the symbol is defined.

## Steps to Reproduce
See the screencap.

## ST3, Anaconda and OS versions
Newest ST3 (3142-dev), Ubuntu 17.04

## ST3 Console Logs
Screencapped
![jump-to-symbol](https://user-images.githubusercontent.com/987055/28848288-abd9b502-7712-11e7-9763-9dac01378aaf.gif)

**Comment by karolyi** [comment]
linking https://github.com/DamnWidget/anaconda/issues/680#issuecomment-319505115

**Comment by DamnWidget** [comment]
I am having a hard time to reproduce this

**Comment by karolyi** [comment]
yep, it's like a corner case, but still, it happens. again, it's the complex codebase I'm using, so I'm not sure what's causing it.
